### ISSUE TITLE
ArgumentError:wrong number of arguments (given 0, expected 1) on Spree::User#send_confirmation_instructions (spree_auth_devise 4.3.0)

### ISSUE THREAD

**Issue by hoshinotsuyoshi** [spree/spree_auth_devise#514]
Hi there,

When updated `4.3.0` today,  I ran into a problem, which does not occur in spree_auth_devise `4.2.0`.

I hope you can check it out : ) 

Thank you.

## Context

* my own spree application environments:
    * ruby 2.7.2
    * spree 4.2.0.rc2
    * devise 4.7.3
    * spree_auth_devise 4.3.0 

* my own spree application:
    * I'm using spree's `testing_support` 's `user` factory .
    * Already set `Spree::Auth::Config[:confirmable] = true` 

### Error:

-------

when I ran `rspec` in my own spree application,  `Spree::User#send_confirmation_instructions` raises error.

`let!(:user) { create(:user) }` results in 

```
ArgumentError:wrong number of arguments (given 0, expected 1) 
```

### Error(a bit more detailed):

------

```
# bundle exec rspec spec/models/spree/user_spec.rb:16
Run options: include {:locations=>{"./spec/models/spree/user_spec.rb"=>[16]}}

Randomized with seed 4998
F

Failures:

  1) Spree::User# (**masked**)
     Failure/Error: let!(:user) { create(:user) }

     ArgumentError:
       wrong number of arguments (given 0, expected 1)
     # /usr/local/bundle/gems/spree_auth_devise-4.3.0/app/models/spree/user.rb:45:in `send_confirmation_instructions'
     # /usr/local/bundle/gems/devise-4.7.3/lib/devise/models/confirmable.rb:183:in `send_on_create_confirmation_instructions'
     # /usr/local/bundle/gems/factory_bot-6.1.0/lib/factory_bot/evaluation.rb:18:in `create'
     # /usr/local/bundle/gems/factory_bot-6.1.0/lib/factory_bot/strategy/create.rb:12:in `block in result'
     # /usr/local/bundle/gems/factory_bot-6.1.0/lib/factory_bot/strategy/create.rb:9:in `tap'
     # /usr/local/bundle/gems/factory_bot-6.1.0/lib/factory_bot/strategy/create.rb:9:in `result'
     # /usr/local/bundle/gems/factory_bot-6.1.0/lib/factory_bot/factory.rb:43:in `run'
     # /usr/local/bundle/gems/factory_bot-6.1.0/lib/factory_bot/factory_runner.rb:29:in `block in run'
     # /usr/local/bundle/gems/factory_bot-6.1.0/lib/factory_bot/factory_runner.rb:28:in `run'
     # /usr/local/bundle/gems/factory_bot-6.1.0/lib/factory_bot/strategy_syntax_method_registrar.rb:28:in `block in define_singular_strategy_method'
     # ./spec/models/spree/user_spec.rb:13:in `block (4 levels) in <top (required)>'
     # /usr/local/bundle/gems/webmock-3.11.1/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

Finished in 0.24938 seconds (files took 31.11 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/models/spree/user_spec.rb:16 # Spree::User# (**masked**)

Randomized with seed 4998

```

## Expected Behavior

`create(:user)` does not raise any error.

## Actual Behavior

`create(:user)` results in 

```
ArgumentError:wrong number of arguments (given 0, expected 1) 
```

## Possible Fix

??

## Steps to Reproduce

1. update spree_auth_devise to 4.3.0 
2. bundle exec rspec

**Comment by damianlegawiec** [comment]
@hoshinotsuyoshi please run `bundle update spree_auth_devise` - this will install `4.3.1` with a fix for this error. Thank you fo reporting it!

**Comment by hoshinotsuyoshi** [comment]
Hi @damianlegawiec , Thank you for your quick fix!

I have tried `spree_auth_devise 4.3.1`.  
But `spree4.2.0.rc2`,  my own spree app,  still seems to have the same problem.

( :eyes:  👍  https://github.com/spree/spree_auth_devise/pull/515 ) 

----

However, I haven't tried spree/spree 's  **master** branch yet.


I'll report back when I've finished trying it out!

As far as I can tell, the spree/spree master branch now requires rails 6.1 (as opposed to 4.2.0.rc2(requires rails 6.0)).  

This rails update process will take some time, so please be patient!

**Comment by damianlegawiec** [comment]
@hoshinotsuyoshi master branch does not require rails 6.1, it's still possible to use it with rails 6.0

**Comment by hoshinotsuyoshi** [comment]
> @hoshinotsuyoshi master branch does not require rails 6.1, it's still possible to use it with rails 6.0

Oh, I'm sorry! You are right. :bow: 


I've just finished trying.

**Comment by hoshinotsuyoshi** [comment]
@damianlegawiec 

Spree master branch ( https://github.com/spree/spree/commit/d3c0e866d59ad243cee6c3184e807e0866a411ed  )  and spree_auth_devise `4.3.1` still seems to have the same problem in my own spree app.  😭  

As backtraces says;

```
  1) Spree::User# (**masked**)
     Failure/Error: let!(:user) { create(:user) }

     ArgumentError:
       wrong number of arguments (given 0, expected 1)
     # /usr/local/bundle/gems/spree_auth_devise-4.3.0/app/models/spree/user.rb:45:in `send_confirmation_instructions'
     # /usr/local/bundle/gems/devise-4.7.3/lib/devise/models/confirmable.rb:183:in `send_on_create_confirmation_instructions'
     # /usr/local/bundle/gems/factory_bot-6.1.0/lib/factory_bot/evaluation.rb:18:in `create'
```

I want to talk about this line:

``/usr/local/bundle/gems/devise-4.7.3/lib/devise/models/confirmable.rb:183:in `send_on_create_confirmation_instructions'``


### devise's model:

https://github.com/heartcombo/devise/blob/v4.7.3/lib/devise/models/confirmable.rb#L52
https://github.com/heartcombo/devise/blob/v4.7.3/lib/devise/models/confirmable.rb#L182-L184

the confirmable.rb  line-183 `#send_confirmation_instructions` is not passed any arguments.

### spree_auth_devise's model:

But spree_auth_devise user.rb 's `#send_confirmation_instructions`  is defined as that receives one argument.

https://github.com/spree/spree_auth_devise/blob/f9ca0e7546f73cbc8e52c8d5fc4b7f38606360fd/app/models/spree/user.rb#L45

----

Therefore, modifying code of controller  does not solve this problem, I think.

**Comment by hoshinotsuyoshi** [comment]
I can confirm now this problem was solved! ( tested on 812ca852f3fdcdfaddc1394d8dda4814588fbfe3 , and spree4.2.0.rc5 ) 😄 

Thank you for your maintaining! @damianlegawiec  @aplegatt
